proc/threads.c

```diff
@@ -1605,19 +1601,33 @@ static int _proc_lockUnlock(lock_t *lock)
 {
 	thread_t *owner = lock->owner;
 	spinlock_ctx_t sc;
-	int ret = 0;
+	int ret = 0, lockPriority;
 
-	lock->owner = NULL;
-	hal_spinlockSet(&threads_common.spinlock, &sc);
+	/* Failsafe - don't crash if we unlock not locked lock */
```

Comment:
IMHO this should result in an error, unlocking not locked lock means that there's something wrong about locking in the program.

Reply:
This if was a dead code anyway, this check is performed outside, removed